src/ServiceControl/MessageRedirects/Api/MessageRedirectsModule.cs

```diff
@@ -0,0 +1,202 @@
+                        FromPhysicalAddress = messageRedirect.FromPhysicalAddress,
+                        ToPhysicalAddress = messageRedirect.ToPhysicalAddress = request.tophysicaladdress.ToLowerInvariant(),
+                    };
+
+                    messageRedirect.LastModifiedTicks = DateTime.UtcNow.Ticks;
+
+                    redirects.Save(session);
+
+                    Bus.Publish(messageRedirectChanged);
+
+                    return HttpStatusCode.OK;
+                }
+            };
+
+            Delete["/redirects/{messageredirectid:guid}/"] = parameters =>
+            {
+                Guid messageRedirectId = parameters.messageredirectid;
+
+                using (var session = Store.OpenSession())
+                {
+                    var redirects = MessageRedirectsCollection.GetOrCreate(session);
+
+                    var messageRedirect = redirects[messageRedirectId];
+
+                    if (messageRedirect == null)
+                    {
+                        return HttpStatusCode.OK;
+                    }
+
+                    redirects.Redirects.Remove(messageRedirect);
+
+                    redirects.Save(session);
+
+                    Bus.Publish<MessageRedirectRemoved>(evt =>
+                    {
+                        evt.MessageRedirectId = messageRedirectId;
+                        evt.FromPhysicalAddress = messageRedirect.FromPhysicalAddress;
+                        evt.ToPhysicalAddress = messageRedirect.ToPhysicalAddress;
+                    });
+                }
+
+                return HttpStatusCode.OK;
+            };
+
+            Head["/redirects"] = _ =>
+            {
+                using (var session = Store.OpenSession())
+                {
+                    var redirects = MessageRedirectsCollection.GetOrCreate(session);
+
+                    return Negotiate
+                        .WithEtagAndLastModified(redirects.ETag, redirects.Redirects.Max(r => new DateTime(r.LastModifiedTicks)))
+                        .WithTotalCount(redirects.Redirects.Count);
+                }
+            };
+
+            Get["/redirects"] = _ =>
+            {
+                using (var session = Store.OpenSession())
+                {
+                    var redirects = MessageRedirectsCollection.GetOrCreate(session);
+
+                    var queryResult = redirects
+                        .Sort(Request)
+                        .Paging(Request)
+                        .Select(r => new
+                        {
+                            r.MessageRedirectId,
+                            r.FromPhysicalAddress,
+                            r.ToPhysicalAddress,
+                            LastModified = new DateTime(r.LastModifiedTicks)
+                        });
+
+                    return Negotiate
+                        .WithModel(queryResult)
+                        .WithEtagAndLastModified(redirects.ETag, queryResult.Max(r => r.LastModified))
```

Comment:
When the collection is empty the Max throws Invalid Operation Expcetion:  Sequence contains no elements